src/drive/mobile/containers/RatingModal.jsx

```diff
@@ -219,29 +148,36 @@ const withBootDelay = (WrappedComponent, showAfterBoots) => {
       hasIncrementedBootCount = true
     }
 
-    dontShowAgain() {
-      this.setState({ prompted: true })
+    dontShowAgain = () => {
+      this.setState(state => ({ ...state, prompted: true }))
     }
 
-    showLater() {
-      this.setState({
+    showLater = () => {
+      this.setState(state => ({
+        ...state,
         bootCount: 0,
         prompted: false
-      })
+      }))
     }
 
     render() {
       const { bootCount, prompted } = this.state
       const visible = prompted === false && bootCount >= showAfterBoots
-      return visible ? (
-        <WrappedComponent
-          dontShowAgain={this.dontShowAgain.bind(this)}
-          showLater={this.showLater.bind(this)}
-          {...this.props}
-        />
-      ) : null
+      return (
+        visible && (
+          <WrappedComponent
+            dontShowAgain={this.dontShowAgain}
+            showLater={this.showLater}
+            {...this.props}
+          />
+        )
+      )
```

Comment:
Why did you change this? As far as I can tell, they do the same, but [it looks to me like the preferred thing to return from a component when you don't want it to render anything](https://reactjs.org/docs/conditional-rendering.html#preventing-component-from-rendering) is `null` rather than `false`.

Reply:
Cannot find the article that proposed to use condition instead of ternary operator. We do this often:

```jsx
return (
  <div>
    {isError && <Error error={error} />}
    Count: {count}
  </div>
);
```

Reply:
Do you agree? Or do you want I change it back?

Reply:
No I don't really mind, I was just curious to know why you changed it since they both work the same.

I know we use ternary's *within* render functions, but in this case it's about the return value of the render function. Not quite the same. The react doc linked above talks about both cases :)